src/app/components/Connect/ConnectSmall.tsx

```diff
@@ -34,6 +37,27 @@ export const ConnectSmall: React.FC<ConnectSmallProps> = ({
   useOnClickOutside(ref, handleClickOutside);
 
   const { coinName, networkName } = getNetworkConfig();
+  const { isApiNormal, isBlocked, apiMessage } = useHealthCheck();
+
+  // Renders the Tooltip describing the reason
+  // why the user might not be able to connect the wallet
+  const renderApiNotAvailableTooltip = () => {
+    if (!isBlocked && isApiNormal) return null;
+
+    return (
+      <>
+        <span
+          className="cursor-pointer text-xs"
+          data-tooltip-id="tooltip-connect"
+          data-tooltip-content={apiMessage}
```

Comment:
I recommend if the error code is `ERROR_GEO_BLOCKED`, then we should always show "The Bitcoin Staking functionality is not available in your region.", regardless of what the error message says. This error code should only be used when someone is geo blocked. If we want to have another error, we can have another code for it.

For example, we could have a general purpose error code, in which case we display the error contained in the message. In the future, if we extend to include more error codes, I expect that we are going to have a big `if/else if/else`, in which on the `if/else if` we check for particular error codes and display custom messages for them and on the `else` we display the error message returned by the API response.

Reply:
@vitsalis right now `errorCode` is `GATEWAY_GEO_BLOCKING`. I used the logic - if it's 451 stick with the `GEO_BLOCK_MESSAGE` and don't use `error?.response?.message` because of two things:
1. I am not sure we are 100% sure about the API contracts and all the responses/errors
2. In this particular case 451 means only one thing - geo blocking
3. All the future API refactorings can happen later on
4. I wouldn't want to add any more different additions to this PR, as it is already very different from what we usually do within the current system and approaches.

So in case of 451, instead of:

```ts
message: error.response?.message || GEO_BLOCK_MESSAGE
```

I am using

```ts
message: GEO_BLOCK_MESSAGE
```

In case of any other errors I will go with 

```ts
message: error.response?.message || error?.message || API_ERROR_MESSAGE
```